### PR TITLE
Design System: Indicator for links opening in new page

### LINKED ISSUE #6254: Indicator for links opening in new page
## Feature Description

Could we add an indicator next to links that open on a new tab/window to show that they are external links? 
For example, the "support" link here, could have a little indicator that lets the user know this will open on a new page. 
image

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

One possible way to do this is looking to the design system `Link` - marking as `isExternal` so that we don't have to track what the internal url is and when that bool is true showing the icon `:after` - that'd let us see how much this would impact the editor and dashboard.

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
We want external links to be more transparent within the app.

## Summary
This request was to add the external link material design icon to all external links:
https://fonts.google.com/icons?selected=Material+Icons+Outlined:launch

## Relevant Technical Choices
NA

## To-do
NA

## User-facing changes
Screen Shot 2021-08-23 at 1 53 31 PM
Screen Shot 2021-08-23 at 1 52 45 PM
Screen Shot 2021-08-23 at 1 52 54 PM

## Testing Instructions
Click through app and see that all links that open in a new tab have this icon.

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6254

### CODE REVIEW

File: packages/design-system/src/components/typography/link/index.js [hunk 1]
```diff
@@ -50,7 +60,28 @@ export const Link = styled.a`
...
+function ConditionalSpanWrapper({ isWrapped, children }) {
+  return isWrapped ? <span>{children}</span> : children;
+}
+ConditionalSpanWrapper.propTypes = {
+  children: PropTypes.node,
+  isWrapped: PropTypes.bool,
+};
+
+export function Link({ children, ...props }) {
+  const isExternalLink = props.target === '_blank';
```
Comment (BrittanyIRL) [inline packages/design-system/src/components/typography/link/index.js:72]: i like this set up!

File: packages/design-system/src/components/typography/link/index.js [hunk 2]
```diff
@@ -50,7 +60,28 @@ export const Link = styled.a`
...
+  return (
+    <StyledAnchor {...props}>
+      <ConditionalSpanWrapper isWrapped={isExternalLink}>
+        {children}
+        {props.target === '_blank' && <StyledLaunch />}
```
Comment (BrittanyIRL) [inline packages/design-system/src/components/typography/link/index.js]: could just reuse `isWrapped`?

**Review by BrittanyIRL — APPROVED** [review]
🚀 🚀  
Works great, tight solution.  Checked Firefox, Safari, Chrome in LTR and RTL. 🤝

**Comment by BrittanyIRL** [comment]
It'd be great to add a storybook example (or just a knob that will let the style toggle), if you have a spare sec.

(also, i'm sure you see this - but looks like some snapshots need a little TLC)

File: packages/design-system/src/components/typography/link/index.js (hunk 2 above)
Comment (maxyinger) [reply packages/design-system/src/components/typography/link/index.js]: ope yup. meant to update that. one sec

**Pull request merged**